runtime/permissions.rs

```diff
@@ -27,6 +27,23 @@ use std::string::ToString;
 use std::sync::atomic::AtomicBool;
 #[cfg(test)]
 use std::sync::atomic::Ordering;
+#[cfg(not(windows))]
+use which::which;
+
+// `which::which` is bugged on Windows, we patch it here.
+// https://github.com/harryfei/which-rs/issues/54
+#[cfg(windows)]
+use which::Result as WhichResult;
+#[cfg(windows)]
+fn which(name: &str) -> WhichResult<PathBuf> {
+  use which::which as which_;
+  which_(name).map(|mut p| {
+    if p.extension().and_then(std::ffi::OsStr::to_str) == Some("EXE") {
+      p.set_extension("exe");
```

Comment:
Paths are case insensitive on windows so we should remove this and fix whatever comparison is being case sensitive to be case insensitive on Windows (and Mac?)

Also, it's not just exe files that are executable on Windows, but whatever is in the PATHEXT environment variable. It seems this dependency handles that for us luckily.